lib/cashaddr.py

```diff
@@ -166,10 +167,11 @@ def decode(address):
     version = decoded[0]
     addr_hash = bytes(decoded[1:])
     size = (version & 0x03) * 4 + 20
+    
     # Double the size, if the 3rd bit is on.
     if version & 0x04:
         size <<= 1
-    if size != len(addr_hash):
+    if size != len(addr_hash) and not rpa_paycode:
```

Comment:
Again this makes me very suspicious -- this is another thing that is well specified by CashAddr.  Why avoid this check in the rpa case?

Can you please point me to the rpa spec?  Can you give me an example of an RPA address so I may try it against the regular bitcoind "Decode" function this is analogous to?

Reply:
I think the way you are supposed to do this is you are supposed to be encoding a version properly into the payload to make the length check succeed here.  

This seems to me like the RPA address spec needs revision if it's producing payloads that don't decode properly in the cashaddr scheme....